### PR TITLE
AD7193 driver extended to support 0/2/4/5 chips

### PR DESCRIPTION
**Pull request by APorumb**
-reworked AD7193 driver to support AD719X family (AD7190/2/3/4/5)

Signed-off-by: Andrei Porumb <andrei.porumb@analog.com>

### CODE REVIEW

File: drivers/adc/ad7193/ad7193.c [hunk 1]
```diff
@@ -93,11 +95,36 @@ int ad7193_init(struct ad7193_dev **device,
...
+	case AD7195:
+		if((reg_val & AD719X_ID_MASK) != AD7195) {
+			goto error_miso;
+		}
+		break;
+	default:
+		return FAILURE;
 	}
 
-
```
Comment (amiclaus) [inline drivers/adc/ad7193/ad7193.c:100]: nit: change unrelated to the commit

File: drivers/adc/ad7193/ad7193.h [hunk 2]
```diff
@@ -199,6 +206,7 @@ struct ad7193_dev {
 	uint8_t			input_mode;
 	uint8_t			buffer;
 	uint8_t     		bpdsw_mode;
+	enum ad719x_adc_ver ver;
```
Comment (amiclaus) [inline drivers/adc/ad7193/ad7193.h]: nit: we usually use `chip_id` when a driver supports a familiy of devices.

**Review by amiclaus — COMMENTED** [review]
Some minor comments on my side.

File: drivers/adc/ad7193/ad7193.c [hunk 3]
```diff
@@ -557,25 +557,25 @@ int ad7193_set_bridge_switch(struct ad7193_dev *dev, uint8_t bpdsw_select)
...
  *                            1 - unipolar operation is selected.
- *@param range     - Gain select bits. These bits are written by the user to select
+ *@param gain      - Gain select bits. These bits are written by the user to select
  *                   the ADC input range.
  *
  * @return SUCCESS in case of success or negative error code.
 *******************************************************************************/
 int ad7193_range_setup(struct ad7193_dev *dev,
-		       uint8_t polarity, enum ad7193_adc_range range)
+		       uint8_t polarity, enum ad719X_adc_gain gain)
```
[the remark this answers is not in the document]
Comment (APorumb) [reply drivers/adc/ad7193/ad7193.c]: They were corrected in the refactoring commit but now I've moved them in the original commit.

File: drivers/adc/ad7193/ad7193.c [hunk 4]
```diff
@@ -319,18 +319,18 @@ int ad7193_reset(struct ad7193_dev *dev)
...
+ *				      AD719X_MODE_PWRDN
+ *				      AD719X_MODE_CAL_INT_ZERO
+ *				      AD719X_MODE_CAL_INT_FULL
+ *				      AD719X_MODE_CAL_SYS_ZERO
+ *				      AD719X_MODE_CAL_SYS_FULL
  * @return SUCCESS in case of success or negative error code.
 *******************************************************************************/
 int ad7193_set_operating_mode(struct ad7193_dev *dev,
-			      enum ad7193_adc_modes opt_mode)
+			      enum ad719X_adc_modes opt_mode)
```
[the remark this answers is not in the document]
Comment (APorumb) [reply drivers/adc/ad7193/ad7193.c]: They were corrected in the refactoring commit but now I've moved them in the original commit.

File: drivers/adc/ad7193/ad7193.h (hunk 2 above)
Comment (APorumb) [reply drivers/adc/ad7193/ad7193.h]: Changed ver to chip_id.

**Comment by APorumb** [comment]
Edit 1:

-renamed ver to chip_id
-removed id tags defines

**Review by buha — APPROVED** [review]

**Review by amiclaus — APPROVED** [review]

**Pull request merged**